internal/devpkg/package.go

```diff
@@ -546,3 +580,30 @@ func (p *Package) StoreName() (string, error) {
 	}
 	return name, nil
 }
+
+// storePath are the constituent parts of
+// /nix/store/<hash>-<name>-<version>
+//
+// This is a helper struct for analyzing the string representation
+type storePathParts struct {
+	hash    string
+	name    string
+	version string
+}
+
+func newStorePathParts(path string) *storePathParts {
```

Comment:
Some names have multiple dashes. You can copy https://github.com/jetpack-io/axiom/blob/main/devbox-cloud/search/internal/nixpkgs/eval.go#L584 which tries to mimic the exact Nix behavior.

Reply:
Good call! will switch